forum/models/model_utils.py

```diff
@@ -431,6 +432,36 @@ def get_user_read_state_by_course_id(
     return {}
 
 
+def get_sort_criteria(sort_key: str) -> Sequence[tuple[str, int]]:
+    """
+    Generate sorting criteria based on the provided key.
+
+    Parameters:
+    sort_key (str): Key to determine sort order ("date", "activity", "votes", "comments").
+
+    Returns:
+    list: List of tuples for sorting, including "pinned" and the relevant field,
+          optionally adding "created_at" if needed.
+    """
+    sort_key_mapper = {
+        "date": "created_at",
+        "activity": "last_activity_at",
+        "votes": "votes.point",
+        "comments": "comment_count",
+    }
+
+    sort_key = sort_key_mapper.get(sort_key, "date")
```

Comment:
if `sort_key` passed to method is `"date"`, then `sort_key_mapper` will return `"created_at"`.
but if `sort_key` does not exists, default is set to `"date"`. Shouldn't it be `"created_at"`.